src/app/shared/components/shared-record-list/shared-record-list.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, Input } from '@angular/core';
+import { Location } from '@angular/common';
+import { ActivatedRoute, Router } from '@angular/router';
+
+import { Record } from 'src/app/shared/interfaces/record';
+import { GenericListComponent } from 'src/app/shared/generics/generic-list/generic-list';
+import { RecordService } from 'src/app/shared/services/records/record.service';
+import { RecordListType } from '../../enums/record-list-type.enum';
+
+@Component({
+  selector: 'app-shared-record-list',
+  templateUrl: './shared-record-list.component.html',
+  styleUrls: ['./shared-record-list.component.css']
+})
+export class SharedRecordListComponent extends GenericListComponent<Record> {
+  @Input() recordListType: RecordListType;
```

Comment:
Mark this as an optional field, `@Input() recordListType?: RecordListType`, and if this field is not defined, then in your `switch` we choose to fall back to retrieve all records.